test/e2e/standard/artifacts.go

```diff
@@ -0,0 +1,120 @@
+package standard
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// collectArtifacts collects individual test artifacts
+// and stores them in ARTIFACTS directory under subdir
+func (sc *SanityChecker) collectArtifacts(namespace, subdir string) error {
+	// namespace = "" is same as --all-namespaces
+	if os.Getenv("ARTIFACTS") != "" {
+		dir := filepath.Join(os.Getenv("ARTIFACTS"), subdir)
+		err := os.MkdirAll(dir, 0755)
+		if err != nil {
+			return err
+		}
+
+		err = sc.getEvents(namespace, dir)
+		if err != nil {
+			return err
+		}
+
+		err = sc.getPodsLogs(namespace, dir)
+		if err != nil {
+			return err
+		}
+
+		err = sc.getPodsStatus(namespace, dir)
+		if err != nil {
+			return err
+		}
```

Comment:
Do we want to wait until we've attempted to collect all of the Events/Logs/PodStatus before erroring out? It might be that we get an error for one but not all three.